cargo-dist/src/cli.rs

```diff
@@ -233,8 +236,33 @@ pub struct InitArgs {
     pub with_json_config: Option<Utf8PathBuf>,
 }
 
+/// Which style(s) of configuration to generate
+#[derive(ValueEnum, Copy, Clone, Debug)]
+pub enum GenerateMode {
+    /// Generate all configuration types (currently: ci)
+    All,
+    /// Generate CI scripts for orchestrating cargo-dist
+    Ci,
+}
+
 #[derive(Args, Clone, Debug)]
-pub struct GenerateCiArgs {}
+pub struct GenerateArgs {
+    /// Which type of configuration to generate
+    pub mode: Vec<GenerateMode>,
+
+    /// Check if the generated output differs from on-disk config without writing it
+    #[clap(long)]
+    #[clap(default_value_t = false)]
+    pub check: bool,
+}
+
+#[derive(Args, Clone, Debug)]
+pub struct GenerateCiArgs {
+    /// Check if the generated output differs from on-disk config without writing it
+    #[clap(long)]
+    #[clap(default_value_t = false)]
+    pub check: bool,
+}
```

Comment:
wait... OH this is your preserving the alias (no idea if there's a less painful way to do it, so respect)

Reply:
Haha, yeah, that's right.